includes/class-wc-ajax.php

```diff
@@ -632,21 +632,22 @@ public static function save_attributes() {
 			ob_start();
 			$attributes = $product->get_attributes( 'edit' );
 			$i          = -1;
+			if( isset( $data['attribute_names'] ) && !empty( $data['attribute_names'] ) ){
```

Comment:
`empty` does not trigger warnings so only that needs to be checked. You can remove the `isset`.

Also, please correct the spacing for code standards. `! empty` and `) {` and `if (`